Makefile

```diff
@@ -0,0 +1,32 @@
+SHELL := bash
+
+PYTHON := $(shell command -v python || command -v python3)
+$(if $(PYTHON),,$(error No python or python3 binary found))
+
+export PYTHONPATH := $(PWD)
+
+TESTS := \
+    test_interpreter \
+    test_vm_compiler \
+    test_barevm \
+
+
+default:
+
+test: $(TESTS)
+
+test_barevm:: barevm/barevm
+
+$(TESTS)::
+	$(PYTHON) tests_full/$@.py
+
+clean:
```

Comment:
This seems really clever :)

How about `git clean -dfx` if you're going to rely on git anyway?

Reply:
I usually try to avoid `git clean` since it removes files like notes.txt that I want to keep around, however I just looked at `git help clean` and learned about the `-X` flag which only removes ignored files. TIL. Perfect!

Reply:
fixed in 35cce6a